proto/directions.proto

```diff
@@ -0,0 +1,94 @@
+package valhalla;
+
+message Waypoint {
+  optional float lat = 1;
+  optional float lon = 2;
+  optional string type = 3;
+  optional uint32 bearing = 4;
+  optional string name = 5;
+  optional string street = 6;
+  optional string side_of_street = 7;
+}
+
+message BoundingBox {
+  optional float min_lat = 1;
+  optional float min_lon = 2;
+  optional float max_lat = 3;
+  optional float max_lon = 4;
+}
+
+message StreetName {
+  optional string text = 1;  // Text of street name, examples: MD 43 East, White Marsh Boulevard 
+  optional bool is_ref = 2;  // true if street name is a ref, examples: true, false
+}
+
+message Step {
+  optional float distance = 1;
+  optional uint32 duration = 2;
+  optional uint32 geometry_index_begin = 3;      // begin geometry inclusive index
+  optional uint32 geometry_index_end = 4;        // end geometry inclusive index
+  optional uint32 maneuver_index_begin = 5;      // the index of the maneuver at the start of the step
+  optional uint32 maneuver_index_end = 6;        // the index of the maneuver at the end of the step
+  repeated StreetName street_names = 7;          // consistent street names for the entire step
+  optional string travel_mode = 8;               // travel mode 
+  optional string travel_mode_type = 9;          // travel type
+  optional string driving_side = 10;             // legal driving side for this step
+}
+  
+message SignElement {
+  optional string text = 1;               // The actual sign element text, example: I 95 North
+  optional bool is_ref = 2;               // true if sign element is a ref, example: true  
+  optional uint32 consecutive_count = 3;  // The frequency of this sign element within a set a consecutive signs
+}
+
+message Sign {
+  repeated SignElement exit_number_elements = 1;  // The list of exit numbers, example: 67B
+  repeated SignElement exit_branch_elements = 2;  // The list of exit branch street names, example: I 95 North
+  repeated SignElement exit_toward_elements = 3;  // The list of exit toward locations, example: New York
```

Comment:
is it possible to call this `exit_destinations` to stay consistent with v5?

Also, is it necessary to say `elements` after each of these, since we already know they are SignElements

Reply:
👍 on removing `elements`

Reply:
thinking about how to rename - exit_numbers and exit_names seem good but exit_branches and exit_towards do not seem correct

Reply:
maybe `is_ref` is too related to OSM - maybe we should change to `is_route_number` to be more generic - since it could be used with other datasets

Reply:
I agree, I find `exit_branches` and `exit_towards` a little confusing. What about `exit_to_streets` or just `exit_streets` and `exit_destinations`?

Reply:
ok so I will update `exit_numbers` and `exit_names`

Reply:
an `exit_toward` can be a place name or a street name that the user is heading toward - it seem like we still need `branch` and `toward` to know the difference. I was trying to stay away from the `exit_to` term so we do not get confused with OSM tag
some suggestions:
`exit_branch_streets` or `exit_onto_streets`
`exit_toward_locations`

Reply:
I like `exit_onto_streets` and `exit_toward_locations` 👌 

you're a great namer!

Reply:
:+1: Thank you @allierowan for improving the interface!